### ISSUE TITLE
[Task] Add ability to open score in previous version of MuseScore in dev builds (add checkbox to DevTools)

### ISSUE THREAD

**Issue by DmitryArefiev** [musescore/MuseScore#20899]
### Task description

This option will help in testing when we want to check regression using a score created in the current MS version

* Add **Disable version checking** checkbox to **DevTools>Settings** (which should be disabled by default)

![Capture](https://github.com/musescore/MuseScore/assets/90187801/a3738522-432a-43a4-b65f-0aabd2e000f8)

**Comment by DmitryArefiev** [comment]
FYI @bkunda 

It will be available only in dev builds

**Comment by cbjeukendrup** [comment]
Another option: 
- bring back the "Open anyway" button, so that users will be able to open 4.4 files in 4.3 again
- but, when re-saving a too-new file in an older version, write a special indication of that into the file. This way, when a user reports a problem with this file, we can see that it may have been caused by opening the file in the wrong version; based on that knowledge we can decide whether or not to spend time on the problem.

(Because as far as I can see, the only reason that we blocked opening new files in older versions was that we didn't want to spend time on problems that arose from this not-really-supported action.)

Having some version history info written in the file is something that seems very useful to me anyway, for troubleshooting purposes.
